source/upgrade/upgrading-mattermost-server.rst

```diff
@@ -1,44 +1,116 @@
 Upgrading Mattermost Server
 ===========================
 
+|all-plans| |self-hosted|
+
+.. |all-plans| image:: ../images/all-plans-badge.png
+  :scale: 30
+  :target: https://mattermost.com/pricing
+  :alt: Available in Mattermost Free and Starter subscription plans.
+
+.. |self-hosted| image:: ../images/self-hosted-badge.png
+  :scale: 30
+  :target: https://mattermost.com/deploy
+  :alt: Available for Mattermost Self-Hosted deployments.
+
 In most cases you can upgrade Mattermost Server in a few minutes, but the upgrade can take longer depending on several factors, including the size and complexity of your installation, and the version that you're upgrading from.
 
+Preparing to Upgrade to the Latest Version
+------------------------------------------
+
+A Mattermost server v6.0 upgrade will run significant database schema changes that can cause a large startup time depending on the dataset size. Zero downtime won't be possible for v6.0, but depending on the efforts made during the migration process, it can be minimized to a large extent. Running queries prior to the upgrade can also save some downtime. However, some queries can still cause full table locking and require Mattermost to be in read-only mode for the duration of the query. 
+
+We recommend that you:
+
+- Set up a maintenance window outside of working hours to mitigate the migration impact. 
+- Keep a backup of your database to ensure you can load a previous database snapshot if necessary.
+- Upgrade your instance of Mattermost to a supported Extended Support Release (ESR) version before attempting to run the v6.0 upgrade.
```

Comment:
Do we want to be more specific here and state 5.37 since 5.31 is coming out of support during this release timeframe?

Reply:
NVM I see it below.